src/main/app/pay/payClient.ts

```diff
@@ -1,54 +1,103 @@
-    const paymentReference = `CMC1$$$${uuid()}$$$AA00$$$${feeCode}`
+  async create (user: User, fees: Fee[], amount: number, returnURL: string): Promise<Payment> {
+    if (!user) {
+      throw new Error('User is required')
+    }
+    if (!fees) {
+      throw new Error('Fees array is required')
+    }
+    if (!amount) {
+      throw new Error('Total fee amount is required')
+    }
+    if (!returnURL) {
+      throw new Error('Post payment redirect URL is required')
+    }
 
-    return request.post({
-      uri: `${payUrl}/users/${user.id}/payments`,
-      body: new PaymentRequest(amount, paymentReference, 'Money Claim issue fee', returnURL),
+    const payment: object = await request.post({
+      uri: baseURL,
+      body: this.preparePaymentRequest(amount, fees),
       headers: {
         Authorization: `Bearer ${user.bearerToken}`,
-        ServiceAuthorization: `Bearer ${this.serviceAuthToken.bearerToken}`
+        ServiceAuthorization: `Bearer ${this.serviceAuthToken.bearerToken}`,
+        'return-url': `${returnURL}`
       }
     })
+    return Payment.deserialize(payment)
   }
 
-  retrieve (user: User, paymentId: string): Promise<Payment> {
-    if (!paymentId) {
-      return Promise.reject(new Error('Payment id must be set'))
+  /**
+   * Retrieves a payment from Reform Payment Hub
+   *
+   * @param user - user who makes a call
+   * @param paymentRef - payment reference number within Reform Payment Hub, generated when payment was created
+   * @returns response all payment details including most recent payment status or undefined when reference does not exist
+   */
+  async retrieve (user: User, paymentRef: string): Promise<PaymentRetrieveResponse | undefined> {
+    if (!user) {
+      throw new Error('User is required')
     }
-
-    return request.get({
-      uri: `${payUrl}/users/${user.id}/payments/${paymentId}`,
-      headers: {
-        Authorization: `Bearer ${user.bearerToken}`,
-        ServiceAuthorization: `Bearer ${this.serviceAuthToken.bearerToken}`
+    if (!paymentRef) {
+      throw new Error('Payment reference is required')
+    }
+    try {
+      const paymentResponse: object = await request.get({
+        uri: `${baseURL}/${paymentRef}`,
+        headers: {
+          Authorization: `Bearer ${user.bearerToken}`,
+          ServiceAuthorization: `Bearer ${this.serviceAuthToken.bearerToken}`
+        }
+      })
+      return plainToClass(PaymentRetrieveResponse, paymentResponse)
+    } catch (err) {
+      if (err.statusCode === HttpStatus.NOT_FOUND) {
+        return undefined
       }
-    })
+      throw err
+    }
+  }
+
+  private preparePaymentRequest (amount: number, fees: Fee[]): object {
+    const caseReference: string = uuid()
+    return {
+      amount: amount,
+      case_reference: caseReference,
+      description: description,
+      service: serviceName,
+      currency: currency,
+      site_id: siteId,
+      fee: fees,
+      ccd_case_number: 'UNKNOWN'
```

Comment:
Is this the value we want hardcoded here?

Reply:
Temporarily yes, we have to send something because CCD API requires some value. In long term we will remove it and stop sending any value.

Reply:
@rlewan Payment-api most latest release requires bit of integration-test at our side which we'll do this week whilst polishing integration work.